R/fill_geographies.R

```diff
@@ -27,21 +27,70 @@ fill_geographies <- function(data) {
     fill_gpprac_geographies()
 }
 
-fill_postcode_geogs <- function(data) {
-  spd <- read_file(get_slf_postcode_path())
+#' Make a postcode lookup for filling to most recent postcodes based on CHI
+#'
+#' @param data A data frame containing chi, postcode, and a date of last contact
+#'
+#' @return A lookup containing CHI numbers and their most recent postcodes
+make_postcode_lookup <- function(data) {
+  postcode_lookup <-
+    data %>%
+    # Get just the chi, postcode and discharge date
+    dplyr::select("chi", "postcode", "record_keydate2") %>%
+    # Get rid of missing chis
+    dplyr::filter(!is_missing(.data$chi)) %>%
```

Comment:
I think if we do a `distinct` right at the top, it will make it even quicker as there will be fewer rows for the `mutate` to work on.
```suggestion
    # Get rid of records with missing CHIs
    dplyr::filter(!is_missing(.data$chi)) %>%
    # Keep one row for each CHI, postcode, and date combination.
    dplyr::distinct("chi", "postcode", "record_keydate2") %>%
```